src/averageAges.js

```diff
@@ -1,40 +1,49 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const specialFilter = (person, condition, value, i, arr) => {
+  switch (condition) {
+    case 'century':
+      return (value ? person.sex === 'm'
+      && Math.ceil(person.died / 100) === value
+        : person.sex === 'm');
+
+    case 'children':
+      return (value ? arr.some(isMother => person.name === isMother.mother)
```

Comment:
Чому в тебе дитина називається isMother?
```suggestion
      return (value
        ? arr.some(isMother => person.name === isMother.mother)
```

Reply:
тут насправді не легко одразу було зрозуміти але саме цим фільтром я шукаю матусь а не дітей! хоча перйменував на більш спірну назву. Якщо ви читаєте ці відповіді, то дай знати а то не впевнений у цьому тому не розписую очму саме матусіі.  ТОді така назва напевне в тему..